library/core/src/time.rs

```diff
@@ -817,14 +876,19 @@ impl Duration {
     #[must_use = "this returns the result of the operation, \
                   without modifying the original"]
     #[inline]
+    #[requires(self.is_safe())]
+    #[ensures(|duration| rhs == 0 || duration.unwrap().is_safe())]
     #[rustc_const_stable(feature = "duration_consts_2", since = "1.58.0")]
     pub const fn checked_div(self, rhs: u32) -> Option<Duration> {
         if rhs != 0 {
             let (secs, extra_secs) = (self.secs / (rhs as u64), self.secs % (rhs as u64));
             let (mut nanos, extra_nanos) = (self.nanos.0 / rhs, self.nanos.0 % rhs);
             nanos +=
                 ((extra_secs * (NANOS_PER_SEC as u64) + extra_nanos as u64) / (rhs as u64)) as u32;
-            debug_assert!(nanos < NANOS_PER_SEC);
+            // TODO: why does this debug assert cause verification to timeout
```

Comment:
It's because division is hard for SAT solvers. How bad is the performance with it? We don't want to change the existing code.

Reply:
I have left it running for over an hour without it finishing. I'll try for longer. What's the max time you would accept? But more fundamentally, I'm confused why this verification would succeed without the debug assert, but fails with the debug_assert when the debug assert is checking essentially the same condition. Do you have any idea why that might be?

Reply:
Good question. @tautschnig might have better insight.

This is not surprising to me though. This reminds me of https://github.com/model-checking/kani/issues/1351

Reply:
Yes, just one extra multiplication (division is turned into a multiplication) can be sufficient to make the problem move from works-in-a-few-minutes to runs-for-several-hours. You might want to change the post-condition so that it is disabled under cfg(kani).

Reply:
I updated the post-condition to just `true` when running Kani, but it still times out for me locally. Not sure what else to try here.

Reply:
I added the `cfg_attr` to the `debug_assert` inside of `checked_div` and Kani was able to complete the proof. Is this addition to the code itself allowed as it doesn't change the behavior when running without Kani?